spec/tom_queue/worker_supervisor_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'tom_queue/worker_supervisor'
+
+describe TomQueue::WorkerSupervisor do
+  let(:supervisor) { described_class.new }
+  describe "#supervise" do
+    it "creates a process with the right name" do
+      supervisor.supervise(as: "worker") { }
+      expect(supervisor.processes.keys).to include "worker1"
+    end
+
+    it "creates a process with the specified behaviour" do
+      my_proc = -> {}
+      supervisor.supervise(as: "worker", &my_proc)
+      expect(supervisor.processes.values).to include my_proc
+    end
+
+    it "can create multiple processes" do
+      supervisor.supervise(as: "worker", count: 2) { }
+      supervisor.supervise(as: "deferred_worker") { }
+      expect(supervisor.processes.keys).to match_array ["worker1", "worker2", "deferred_worker1"]
+    end
+  end
+
+  describe "#before_fork" do
+    it "assigns the before_fork task" do
+      before_fork_task = -> { }
+      supervisor.before_fork = before_fork_task
```

Comment:
We're setting the fork hooks with assignment instead of the snazzy suggestion of `supervisor.after_fork {}` just because it was easier to implement. We can rework it if it's important.

Reply:
Nah, that's cool!